lib/pbench/test/unit/server/query_apis/test_datasets_delete.py

```diff
@@ -102,6 +103,40 @@ def fake_bulk(
 
         monkeypatch.setattr("elasticsearch.helpers.streaming_bulk", fake_bulk)
 
+    def unelastic(self, monkeypatch):
+        """Cause a failure if the bulk action helper is called.
+
+        Args:
+            monkeypatch: The monkeypatch fixture from the test case
+
+        Raises:
+            an unexpected Exception if called
+        """
+
+        def fake_bulk(
+            elastic: elasticsearch.Elasticsearch,
+            stream: Iterator[dict],
+            raise_on_error: bool = True,
+            raise_on_exception: bool = True,
+        ):
+            """
+            Helper function to mock the Elasticsearch helper streaming_bulk API,
+            which will validate the input actions and generate expected responses.
+
+            Args:
+                elastic: An Elasticsearch object
+                stream: The input stream of bulk action dicts
+                raise_on_error: indicates whether errors should be raised
+                raise_on_exception: indicates whether exceptions should propagate
+                    or be trapped
+
+            Raises:
+                Exception
+            """
+            raise Exception("We aren't allowed to be here")
```

Comment:
Insert here the usual caveat about the fact that raising exceptions inside mocks is an unreliable way to cause a test (or anything else) to fail....  (I _would_ recommend using [`pytest.fail()`](https://docs.pytest.org/en/7.1.x/reference/reference.html#id9) instead of `raise`, but, apparently, [it just raises an exception](https://docs.pytest.org/en/7.1.x/_modules/_pytest/outcomes.html#fail).  😝)